draft-thomson-train-protocol.md

```diff
@@ -227,6 +255,26 @@ An endpoint that receives and discards a packet when there are no valid packets
 in the datagram SHOULD ignore any rate limit signal.  Such a datagram might be
 entirely spoofed.
 
+# The TRAIN BANDWIDTH frame {#tbwf}
+
+The rate limit designated by one of the reserved version values indicates
+the rate limit for packets sent in that direction of transport. The endpoint
+receiving the indication SHOULD forward it to the peer by posting the
+value in a TRAIN_BANDWIDTH frame.
```

Comment:
Let's discuss the TP or not in a different PR.

Reply:
My opinion is that QUIC shouldn't need to pass this signal.  Applications will generate their own signaling based on this, as necessary.  For instance, in a streaming video app, the bandwidth cap will determine how the client requests new video chunks.  There is no direct need to tell the server what to do.

I can't see a generic signaling mechanism being used in most of the cases we're talking about, so I'd rather not define one.

Reply:
The signal can be useful as a hint for congestion control. This is indeed debatable, since the signal indicates a theoretical maximum, not the actual value, and CC will still have to discover the available bandwidth. But BBR, for example, has special code to detect on-path pacers. If the limit is 20 Mbps and the CC algorithms gets it slightly too high, say 21 Mbps, this can result in lots of errors. Knowing the limit would avoid such failure modes. But then again, trusting the limit could result in unnecessary slowdowns.